### ISSUE TITLE
Support `gs_multi_send` in the `safe-apps-provider`

### ISSUE THREAD

**Issue by mikhailxyz** [safe-global/safe-apps-sdk#373]
We are migrating the wallet-connect safe app to the safe-apps-provider:
https://github.com/safe-global/safe-react-apps/pull/537

The only thing the `safe-apps-provider` lacks is the support for the OG `gs_multi_send` method for batching transactions. Supporting this method has some complexity because under the hood, the `safe-apps-provider` stores a fake transaction that includes the recipient address, data, etc. Since the transactions passed to the `gs_multi_send` method will be encoded and sent to the multisend contract, we need somehow to get the encoded data and the address of the multisend contract.

I see two options to achieve this:
1. Replicate the encoding/address retrieval logic from our interfaces in the `safe-apps-provider`
2. Change the `txs.send` return type to return the whole safe transaction object

I lean towards 2)

**Comment by PaulRBerg** [comment]
Is this [MultiSend](https://etherscan.io/address/0x8d29be29923b68abfdd21e541b9374737b49cdad) that you referred to here, @mmv08?

**Comment by mmv08** [comment]
> Is this the [MultiSend](https://etherscan.io/address/0x8d29be29923b68abfdd21e541b9374737b49cdad) contract that you referred to here, [@mmv08](https://github.com/mmv08)?

Yes and no. When I created the issue, that likely would've been the contract we'd use. Today the contract of choice would've been the MultiSendCallOnly contract because it doesn't allow nested delegatecalls and is easier to reason about the security of the transaction. The addresses it deployed to can be found [here](https://github.com/safe-global/safe-deployments/blob/d72ccc6c4cf11b2085153ca1a4a24dff738ba3ad/src/assets/v1.4.1/multi_send_call_only.json) and the source code is available [here](https://github.com/safe-global/safe-smart-account/blob/21dc82410445637820f600c7399a804ad55841d5/contracts/libraries/MultiSendCallOnly.sol)

I don't think this issue is relevant given that we support [ERC-5792](https://eips.ethereum.org/EIPS/eip-5792) nowadays

**Comment by mmv08** [comment]
> Thanks for the quick answer [@mmv08](https://github.com/mmv08)!
> 
> Related issue: https://github.com/safe-global/safe-deployments/blob/d72ccc6c4cf11b2085153ca1a4a24dff738ba3ad/src/assets/v1.4.1/multi_send_call_only.json

is the link correct?
